frontend/app/src/main/java/dk/scheduling/schedulingfrontend/device/DeviceState.kt

```diff
@@ -0,0 +1,27 @@
+package dk.scheduling.schedulingfrontend.device
+
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+
+enum class DeviceState {
+    Active,
+    Scheduled,
+    Inactive,
+}
+
+fun getDeviceState(deviceOverview: DeviceOverview): DeviceState {
+    val event = deviceOverview.event ?: return DeviceState.Inactive
+
+    if (event.startTime.isAfter(LocalDateTime.now())) {
```

Comment:
Save local time in a variable instead of calling it 3+ times through this function, it should not make a difference in the correctness.